src/nxdoc/nuxeo-server/administration/variables-and-configuration/variables-and-configuration-pattern.md

```diff
@@ -0,0 +1,300 @@
+---
+title: Variables and Configuration Pattern
+review:
+    comment: ''
+    date: '2020-12-14'
+    status: ok
+tree_item_index: 90
+toc: true
+---
+
+{{! excerpt}}
+This page provides clarifies all options available in the current Nuxeo framework.
+{{! /excerpt}}
+Variable Options
+
+## Glossary
+
+Let's review first the main concepts:
+- **Environment variables**: variables set in the OS that cannot be changed at runtime
+- **System properties**: variables set on the Java command line using the `-Dpropertyname=value` syntax, that can be added at runtime using `System#setProperties` or `System.getProperties#load` method, retrieved through the various `System#getProperty` methods.
+- **Nuxeo configuration properties**: variables set in the [nuxeo.conf]({{page version='' space='nxdoc' page='configuration-parameters-index-nuxeoconf'}}) file.
+- **Nuxeo template properties**: variables set via Nuxeo configuration templates properties files loaded via Java properties utilities.
+
+
+## Variables Retrieval
+
+Variables are retrieved and set at different phases of the Nuxeo application startup, the rest of the document will focus on:
+
+- **Application startup**: in this phase, [ConfigurationGenerator](https://github.com/nuxeo/nuxeo/blob/v11.4.31/modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationGenerator.java) logics will retrieve and aggregate some variables from all variables and properties described in the glossary. These variables will be taken into account for the Nuxeo Runtime startup.
+- **Nuxeo Runtime startup**: in this phase, the Nuxeo Runtime Framework will resolve XML configuration files for Nuxeo Runtime components: these files can reference variables that will be expanded, and can then be retrieved from Java code.
+
+## Application Startup
+
+Some subjects were dismissed for the purpose of this document:
+
+- Properties that control where the nuxeo home is set, or where the templates directory is set, or where logs are set, for instance, as these are handled specifically,
+- Specific logics around database choice and initialization,
+- Cryptographic-related logics.
+
+### Pre-processing Initialization
+
+Java class [ConfigurationGenerator](https://github.com/nuxeo/nuxeo/blob/v11.4.31/modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationGenerator.java) holds [pre-processing](https://github.com/nuxeo/nuxeo/blob/v11.4.31/modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationGenerator.java#L572) logics that will impact the Runtime Properties Initialization. As a general rule, in case of duplicate keys for properties loaded during the different steps, the **last** loaded value will **override**  the previous one:
```

Comment:
Usage of v11.4.31 instead of master to be discussed with @manonlumeau (especially if this page is taken into account for the LTS version).
Also I'm pretty sure the 11.4 promoted version was actually 11.4.42.

Reply:
Updated links to use v11.4.42 --> pertinence of keeping those still open for discussion (especially for maintenance-related issues)

Reply:
Hi @atchertchian! For this remark, this page will be used for master as done on the master branch, do I keep the version like this or does it need to be updated? 
We will probably need to put a different one on the `2021` branch. 
Thanks!

Reply:
Replaced version by `master` as discussed @manonlumeau